src/qibo/models/error_mitigation.py

```diff
@@ -302,6 +306,27 @@ def sample_training_circuit_cdr(
     return sampled_circuit
 
 
+def _curve_fit(
+    backend, model, params, xdata, ydata, lr=1, max_iter=int(1e2), tolerance_grad=1e-5
+):
+    if backend.name == "pytorch":
+        loss = lambda pred, target: backend.np.mean((pred - target) ** 2)
+        optimizer = backend.np.optim.LBFGS(
```

Comment:
LBFGS should be a gradient-based optimizer, thus I am not sure it will work well in case we use shots. 
It is true here we just fit the plot of noisy vs exact, so it is probably fine,  but, just in case, I would use Powell instead.

Reply:
pytorch doesn't seem to have any implementation for the Powell method unfortunately. In any case here we are just fitting a given function on some `(x, y)` points which should be independent on how the points were generated, with shots or not.

Reply:
> In any case here we are just fitting a given function on some `(x, y)` points which should be independent on how the points were generated, with shots or not.

Yes indeed, you are right @BrunoLiegiBastonLiegi.

> It should be documented somewhere that the optimizer for torch is different from the others.

I think all the backends are still supporting our optimizers, but yes, I would definitely mention that each framework (in particular QML ones) can work with their own optimizers too and in some cases this choice is necessary (like here).